test/libmpv_test_track_selection.c

```diff
@@ -0,0 +1,193 @@
+/*
+ * This file is part of mpv.
+ *
+ * mpv is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "libmpv_common.h"
+
+static void check_string(const char *property, const char *expect)
+{
+    char *result_string;
+    check_api_error(mpv_get_property(ctx, property, MPV_FORMAT_STRING, &result_string));
+    if (strcmp(expect, result_string) != 0)
+        fail("String: expected '%s' but got '%s'!\n", expect, result_string);
+    mpv_free(result_string);
+}
+
+static void wait_for_file_load(void)
+{
+    bool loaded = false;
+    while (!loaded) {
+        mpv_event *event = wrap_wait_event();
+        switch (event->event_id) {
+        case MPV_EVENT_FILE_LOADED:
+            loaded = true;
+            break;
+        }
+    }
+}
+
+static void test_track_selection(char *file, char *path)
+{
+    FILE *fp = fopen(path, "r");
+    if (!fp)
+        fail("Test file, '%s', was not found!\n", path);
```

Comment:
missing fclose.
consider [`access(3)`](https://man.archlinux.org/man/access.3p) (unless mingw somehow doesn't have that)

Reply:
Changed to `access`